### PR TITLE
Include virtualservice mirror destinations in analysis

### PR DESCRIPTION
**Pull request by sushicw**
Fixes https://github.com/istio/istio/issues/17999

Mirror destinations should be checked for correctness just like route destinations.

### CODE REVIEW

**Comment by sushicw** [comment]
> Not sure if I understand.
> 
> Does it mean that if this PR gets in I cannot mirror requests to an external host anymore?

[As I understand it](https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#Destination), the Destination specified under `mirror` is the same as the Destination(s) specified under routes. So yes, it could be external, but, just as with routing you'd need a ServiceEntry defining the external host. 

This PR just applies the same checking for a valid Destination to what's specified in `mirror`, where it was previously not getting validated.

File: galley/pkg/config/analysis/analyzers/analyzers_test.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ var testGrid = []testCase{
 		expected: []message{
 			{msg.ReferencedResourceNotFound, "VirtualService reviews-bogushost.default"},
 			{msg.ReferencedResourceNotFound, "VirtualService reviews-bookinfo-other.default"},
+			{msg.ReferencedResourceNotFound, "VirtualService reviews-mirror-bogushost.default"},
```
Comment (selmanj) [inline galley/pkg/config/analysis/analyzers/analyzers_test.go]: Would it be useful to have a test case where the mirrored host exists and we ensure no message is emitted? If you think there's no possible way such a test might fail, then I'm fine leaving this as is.

Comment (sushicw) [reply galley/pkg/config/analysis/analyzers/analyzers_test.go]: It's probably overkill but it's also cheap, I'll do it.

**Comment by jwendell** [comment]
> > Not sure if I understand.
> > Does it mean that if this PR gets in I cannot mirror requests to an external host anymore?
> 
> [As I understand it](https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#Destination), the Destination specified under `mirror` is the same as the Destination(s) specified under routes. So yes, it could be external, but, just as with routing you'd need a ServiceEntry defining the external host.

You don't need a ServiceEntry if the policy is set to ALLOW_ANY.

**Comment by sushicw** [comment]
> > > Not sure if I understand.
> > > Does it mean that if this PR gets in I cannot mirror requests to an external host anymore?
> > 
> > 
> > [As I understand it](https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#Destination), the Destination specified under `mirror` is the same as the Destination(s) specified under routes. So yes, it could be external, but, just as with routing you'd need a ServiceEntry defining the external host.
> 
> You don't need a ServiceEntry if the policy is set to ALLOW_ANY.

Created https://github.com/istio/istio/issues/18764 to track respecting `outboundTrafficPolicy.mode`.

**Pull request merged**